plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -494,6 +502,21 @@ private static DeltaLakeColumnMetadata mapColumn(TypeManager typeManager, JsonNo
         return new DeltaLakeColumnMetadata(columnMetadata, fieldName, fieldId, physicalName, physicalColumnType);
     }
 
+    private static void verifyTypeChange(JsonNode typeChange)
+    {
+        String fromType = typeChange.get("fromType").asText();
+        String toType = typeChange.get("toType").asText();
+
+        switch ("%s -> %s".formatted(fromType, toType)) {
+            case "byte -> short":
```

Comment:
this is a hack we used in tests, but i wouldn't use same pattern in prod code.
let's have direct if on from and to types.